### DISCUSSION TITLE
COTS UE not seeing gNB using USRP B210 (srsRAN Project)

### DISCUSSION THREAD

**Discussion by aldalahmeh** [srsran/srsRAN_Project#134] [General Help]
I have followed the instruction [here](https://docs.srsran.com/projects/project/en/latest/tutorials/source/cotsUE/source/index.html#external-clock), but could not get my COTS (one plus phone) to the gNB. Here are my config files
[amf.txt](https://github.com/srsran/srsRAN_Project/files/12095514/amf.txt)
[upf.txt](https://github.com/srsran/srsRAN_Project/files/12095516/upf.txt)
[gNB.txt](https://github.com/srsran/srsRAN_Project/files/12095518/gNB.txt)

I know the doc states it might be a synchronization issue, since the B210 doesn't have a GPSDO source. However, my colleagues, managed to pull it off using a BladeRF with no sync, as it in my case. 

On an another note, I am not comfortable with the transmission power levels I'mm getting here. See the pics below before and after I start the gNB. Although the peak power reaches -96dBm but the average is still around -125dBm. I tried to increase the Tx gain in the gNB config file but it did not matter much.

![before_gNB](https://github.com/srsran/srsRAN_Project/assets/109239143/37c0da7c-40e2-434e-9891-6599927df23d)
![after_gNB](https://github.com/srsran/srsRAN_Project/assets/109239143/65e02e81-be49-42d9-a902-c9b264bfef92)

**Comment by herlesupreeth** [comment]
@aldalahmeh please use the below values for gain and give it a try

```
  tx_gain: 80                                                   # Transmit gain of the RF might need to adjusted to the given situation.
  rx_gain: 40                                                   # Receive gain of the RF might need to adjusted to the given situation.
```

**Reply by aldalahmeh** [reply]
Unfortunately, didn't work. However, it turns out to be a uhd configuration. I create a flowgraph in GNURadio for an AM transmitter with default USRP setting and got the same low power issue. The flowgrqaph pic is attached below. However, when I changed the RF options [Ch0 Fain Type: Normalized], I got up to -67dBm, see pic below. The question now is how to change such setting in the gNB conf file?

![image](https://github.com/srsran/srsRAN_Project/assets/109239143/7e2665c1-513c-4c3b-b4ed-46a46491758c)
![image](https://github.com/srsran/srsRAN_Project/assets/109239143/4b276f01-2fc5-4b10-aeda-b409834799aa)

**Reply by herlesupreeth** [reply]
I dont think its possible from config file of gNB since there are no device args defined for that in UHD (check below page for list of device args)

https://files.ettus.com/manual/page_configuration.html

**Reply by aldalahmeh** [reply]
So how to change the power settings and still run the gnb command?

**Reply by herlesupreeth** [reply]
I will let others answer to this as I dont know how to do those changes.

**Reply by pgawlowicz** [reply]
@aldalahmeh any update on this issue?

**Comment by aldalahmeh** [comment]
It turned out that at that time and place the interference on that band was very high. I got it working somewhere else. Thanks for the followup.

**Comment by Alt-Shivam** [comment]
Hi @aldalahmeh 
The console you're using looks interesting.
Can you share a link or something, to set up this PlutoSDR console?
can it be used with a USRP?

Thanks

**Reply by aldalahmeh** [reply]
I used a B210 with srsRAN code (see below)
[https://www.srslte.com/](url)

**Reply by Alt-Shivam** [reply]
Hi @aldalahmeh 
The link says "Page not found"
